component/src/main/java/me/androidbox/component/agenda/VisitorFilter.kt

```diff
@@ -0,0 +1,102 @@
+package me.androidbox.component.agenda
+
+import androidx.annotation.StringRes
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import me.androidbox.component.R
+import me.androidbox.component.general.TaskButton
+import me.androidbox.component.ui.theme.*
+import java.util.*
+
+@Composable
+fun VisitorFilter(
+    modifier: Modifier = Modifier,
+    listOfVisitorType: List<VisitorType>,
+    selectedVisitorType: VisitorType,
+    onSelectedTypeClicked: (VisitorType) -> Unit) {
+
+    Column(modifier = modifier) {
+        Text(
+            text = stringResource(R.string.visitors),
+            fontWeight = FontWeight.Bold,
+            fontSize = 20.sp,
+            color = MaterialTheme.colorScheme.agendaBodyTextColor)
+
+        Spacer(modifier = Modifier.height(36.dp))
+
+        Row(modifier = Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.SpaceBetween) {
+            listOfVisitorType.forEach { visitorType ->
+                val backgroundColor = getBackgroundColor(selectedVisitorType, visitorType)
+                val textColor = getTextColor(selectedVisitorType, visitorType)
```

Comment:
I personally don't find these extra functions necessary for a simple if, as long as they're not reused